frontend/app/src/App.tsx

```diff
@@ -1323,17 +1323,27 @@ export class App extends PureComponent<Props, State> {
 
     const usingCustomTheme = !isPresetTheme(this.props.theme.activeTheme)
     if (themeInput) {
-      const customTheme = createTheme(CUSTOM_THEME_NAME, themeInput)
-      // For now, users can only add one custom theme.
-      this.props.theme.addThemes([customTheme])
+      const customThemes = createCustomThemes(themeInput)
+
+      // Add the themes to the theme manager
+      this.props.theme.addThemes(customThemes)
 
       const userPreference = getCachedTheme()
       if (userPreference === null || usingCustomTheme) {
         // Update the theme to be customTheme either if the user hasn't set a
         // preference (developer-provided custom themes should be the default
         // for an app) or if a custom theme is currently active (to ensure that
         // we pick up any new changes to it).
-        this.setAndSendTheme(customTheme)
+        if (customThemes.length > 1) {
```

Comment:
Can there be more than 2 themes? If so, is it always [light theme, dark theme, ...other themes]. Just want to make sure this is codifed well, so that all this implicit knowledge does not get lost across files.

Not a blocker, but have we considered a simple class that takes in an array, and then has methods like `getLightTheme`, `getDarkTheme`, `getOtherThemes` (it's probably overengineering, but just a thought).